### PR TITLE
Feature/boas 1037 case stage

### PR DESCRIPTION
**Pull request by HadrianHughesKC**
## Describe your changes

* feat(api/applications): add stage to Swagger spec
* feat(packages/applications): add type for case stage
* feat(api/applications): enable updating case stage
* feat(web/applications): render the case status in project information
* feat(web/applications): render a template for editing case stage
* feat(api/applications): create an endpoint to fetch all case stages
* feat(web/applications): render radio button layout for editing stage
* feat(web/applications): perform API call to update case stage
* chore(web/applications): add a validator for the stage parameter
* feat(web/applications): filter out the draft option from case stage list
* feat(web/applications): add info text below case stage radio buttons

Tested locally.

## Issue ticket number and link

[BOAS-1037](https://pins-ds.atlassian.net/browse/BOAS-1037)

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [x] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

[BOAS-1037]: https://pins-ds.atlassian.net/browse/BOAS-1037?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/api/src/server/applications/application/application.validators.js [hunk 1]
```diff
@@ -103,6 +103,28 @@ const validateExistingMapZoomLevel = async (value) => {
...
+	const isValid = [
+		'pre_application',
+		'draft',
+		'acceptance',
+		'pre_examination',
+		'examination',
+		'recommendation',
+		'decision',
+		'post_decision',
+		'withdrawn'
```
Comment (HadrianHughesKC) [inline apps/api/src/server/applications/application/application.validators.js:120]: Normally I would declare `ApplicationStageType` like

```typescript
const applicationStages = ['draft', 'examination', 'etc'] as const;
type ApplicationStageType = typeof applicationStages[number];
```

but it's not possible to do this in a `.d.ts` file. Not ideal to have a static list of the case stages in multiple places.

File: apps/web/src/server/applications/common/components/form/form-case.component.js [hunk 2]
```diff
@@ -31,6 +34,52 @@ export async function caseNameAndDescriptionData(request, locals) {
...
+ * Format stage property for stage edit page
+ *
+ * @param {import('express').Request} request
+ * @param {Record<string, any>} locals
+ * @returns {Promise<ApplicationsCreateCaseStageProps>}
+ * */
+export async function caseStageData(request, locals) {
+	/** @type {{name: string, displayNameEn: string}[]} */
+	const allStages = await getAllCaseStages();
+	const stages = allStages.filter((stage) => stage.name !== 'draft');
```
Comment (lutamKC) [inline apps/web/src/server/applications/common/components/form/form-case.component.js:47]: allStages could be undefined or an error object from the api. consider using a "?" or a `allStage||[]`

File: apps/web/src/server/applications/common/services/entities.service.js [hunk 3]
```diff
@@ -33,6 +33,10 @@ export const getAllRegions = () => {
 	return get(`applications/region`);
 };
 
+export const getAllCaseStages = () => {
+	return get('applications/case-stage');
```
Comment (lutamKC) [inline apps/web/src/server/applications/common/services/entities.service.js:37]: this function doesn't handle errors. it's fine but you need some validation in the file above

**Review by lutamKC — DISMISSED** [review]

File: apps/web/src/server/applications/common/components/form/form-case.component.js (hunk 2 above)
Comment (HadrianHughesKC) [reply apps/web/src/server/applications/common/components/form/form-case.component.js:55]: 👍 That's done

File: apps/web/src/server/applications/common/services/entities.service.js (hunk 3 above)
Comment (HadrianHughesKC) [reply apps/web/src/server/applications/common/services/entities.service.js]: 👍 Added a try-catch around the function consuming `getAllCaseStages`.

**Review by lutamKC — APPROVED** [review]

File: apps/api/src/server/repositories/case.repository.js [hunk 4]
```diff
@@ -301,6 +304,14 @@ const updateApplicationSansRegionsRemoval = ({
 						}
 					}
 				}),
+			...(caseStatus && {
+				CaseStatus: {
+					updateMany: {
```
Comment (DavidJWhiteDev) [inline apps/api/src/server/repositories/case.repository.js:309]: I thought that CaseStatus kept an audit history of all the status changes on a case, with the valid = 1 indicating the latest record. .  This code updates all status records for that case to the new value.

**Pull request merged**